src/app.cpp

```diff
@@ -74,6 +75,10 @@ void transform(
 	else {
 		// tecmp packet
 		while (res == 0) {
+			if (ignore_replay_data && header.message_type == TECMP_TYPE_REPLAY_DATA) {
```

Comment:
You don't have to do this check inside the while loop, replay data don't get mixed with other message types

Reply:
You mean that the check should be done inside `tecmp_next` method?

Reply:
it should be done just before the `while (res == 0)` check